src/utl/src/prometheus/metrics_server.cpp

```diff
@@ -0,0 +1,156 @@
+
+// Copyright 2025 Google LLC
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#include "utl/prometheus/metrics_server.h"
+
+#include <boost/asio.hpp>
+#include <boost/beast.hpp>
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <thread>
+
+#include "utl/Logger.h"
+
+namespace utl {
+
+PrometheusMetricsServer::~PrometheusMetricsServer()
+{
+  shutdown_ = true;
+
+  // Make a dummy connection to unblock the accept().
+  if (is_ready_) {  // Only connect if the server was actually started.
+    try {
+      boost::asio::io_context
+          io_context;  // Use a separate io_context for the connection.
+      boost::asio::ip::tcp::socket socket(io_context);
+      boost::asio::ip::tcp::endpoint endpoint(
+          boost::asio::ip::address::from_string("127.0.0.1"), port_);
+      socket.connect(endpoint);         // This will unblock the accept().
+    } catch (const std::exception& e) { /*Do nothing, we're dying*/
+    }
+  }
+  worker_thread_.join();
+}
+
+std::string SnapshotPrometheusMetrics(Registry* registry)
```

Comment:
local functions should be in an anonymous namespace https://google.github.io/styleguide/cppguide.html#Internal_Linkage